gym/envs/robot_locomotion_group/drake/pusher_slider/pusher_slider_env.py

```diff
@@ -0,0 +1,253 @@
+        assert self._sim_diagram.is_finalized()
+        if context is None:
+            context = self._simulator.get_context()
+        context = self._diagram.GetMutableSubsystemContext(self._sim_diagram, context)
+        mbp_context = self._sim_diagram.GetMutableSubsystemContext(
+            self._sim_diagram.mbp, context)
+        obs = dict()
+        obs["time"] = context.get_time()
+        obs["pusher"] = dict()
+        obs["pusher"]["position"] = np.copy(self._sim_diagram.mbp.GetPositions(
+            mbp_context, self._pusher_model_id))
+        obs["pusher"]["velocity"] = np.copy(self._sim_diagram.mbp.GetVelocities(
+            mbp_context, self._pusher_model_id))
+        q = np.copy(self._sim_diagram.mbp.GetPositions(mbp_context,
+                                                       self._slider_model_id))
+        v = np.copy(self._sim_diagram.mbp.GetVelocities(mbp_context,
+                                                        self._slider_model_id))
+        obs["slider"] = dict()
+        obs["slider"]["position"] = {'translation': q[4:], 'quaternion': q[0:4], 'raw': q}
+        obs["slider"]["velocity"] = {'linear': v[3:], 'angular': v[0:3], 'raw': v}
+        if self._config["rgbd_sensors"]["enabled"]:
+            obs["images"] = self._sim_diagram.get_image_observations(context)
+        return obs
+
+    def pusher_slider_on_table(self, context=None):
+        if context is None:
+            context = self._simulator.get_context()
+        context = self._diagram.GetMutableSubsystemContext(self._sim_diagram, context)
+        mbp_context = self._sim_diagram.GetMutableSubsystemContext(
+            self._sim_diagram.mbp, context)
+
+        table_dim = np.array(self._config["table"]["size"])
+        q_pusher = self._sim_diagram.mbp.GetPositions(mbp_context,
+                                                      self._pusher_model_id)
+        q_slider = self._sim_diagram.mbp.GetPositions(
+            mbp_context, self._slider_model_id)[4:6]
+
+        tol = 0.03
+        high_edge = table_dim[:2]/2.0 - tol
+        low_edge = -high_edge
+
+        return ((low_edge < q_pusher) & (q_pusher < high_edge) &
+                (low_edge < q_slider) & (q_slider < high_edge)).all()
+
+    def step(self, action, dt=None):
+        assert self._sim_diagram.is_finalized()
+        assert len(action) == 2
+        if dt is None:
+            dt = self._step_dt
+
+        # the time to advance to
+        t_advance = self._simulator.get_context().get_time() + dt
+        context = self._simulator.get_mutable_context()
+
+        # set the value of the PID controller input port
+        pid_setpoint = np.concatenate((np.zeros(2), action))
+        self._pid_desired_state_port.FixValue(context, pid_setpoint)
+
+        # simulate and take observation
+        self._simulator.AdvanceTo(t_advance)
+        obs = self.get_observation()
+
+        reward = 0.
+        done = self.pusher_slider_on_table()
```

Comment:
This line is a bit strange. (When I run `python3 examples/agents/random_agent.py PusherSlider-v0`, it resets too quickly because it receives `done` after a single timestep.) Is this the intended behavior? Consider setting this to `None`?

Reply:
Done.